### PR TITLE
docs(upgrade) upgrade path for 2.1.0

### PR DESCRIPTION
**Pull request by hishamhm**

### CODE REVIEW

File: UPGRADE.md [hunk 1]
```diff
@@ -37,6 +37,158 @@ are terminated. In this way, Kong will serve new requests via the new
...
+previous release, so you will need to rebuild them with the latest patches.
+
+In order to use all Kong features, including the new
+dynamic upstream keepalive behavior, the required OpenResty version is
+[1.15.8.3](http://openresty.org/en/changelog-1015008.html) and the
+the set of [OpenResty
+patches](https://github.com/Kong/kong-build-tools/tree/master/openresty-build-tools/openresty-patches)
+included has changed, including the latest release of
+[lua-kong-nginx-module](https://github.com/Kong/lua-kong-nginx-module).
+Our [kong-build-tools](https://github.com/Kong/kong-build-tools)
```
Comment (gszr) [inline UPGRADE.md:70]:
```suggestion
Our [openresty-build-tools](https://github.com/Kong/kong-build-tools/tree/master/openresty-build-tools)
```

File: UPGRADE.md [hunk 2]
```diff
@@ -37,6 +37,158 @@ are terminated. In this way, Kong will serve new requests via the new
...
+5. From your 2.1.0 cluster, run: `kong migrations finish`.
+   From this point on, it will not be possible to start
+   nodes in the old cluster pointing to the same datastore anymore. Only run
+   this command when you are confident that your migration
+   was successful. From now on, you can safely make Admin API
+   requests to your 2.1.0 nodes.
+
+**Cassandra**
+
+Due to internal format changes, the Cassandra database for Kong 2.1.0 is
```
Comment (gszr) [inline UPGRADE.md:156]: It is unclear whether this section is describing a Kong upgrade only or a Cassandra version upgrade too. Might be helpful to clarify that.

File: UPGRADE.md [hunk 3]
```diff
@@ -37,6 +37,158 @@ are terminated. In this way, Kong will serve new requests via the new
 configuration, without dropping existing in-flight connections.
 
 
+## Upgrade to `2.1.0`
+
+Kong adheres to [semantic versioning](https://semver.org/), which makes a
+distinction between "major", "minor" and "patch" versions. The upgrade path
```
Comment (gszr) [inline UPGRADE.md:43]:
```suggestion
distinction between "major", "minor", and "patch" versions. The upgrade path
```

File: UPGRADE.md (hunk 1 above)
Comment (gszr) [reply UPGRADE.md:70]: (pointing to `kong-build-tools` root might be a little confusing to users unfamiliar with those tools, as that is broader than just building openresty)

File: UPGRADE.md [hunk 4]
```diff
@@ -37,6 +37,158 @@ are terminated. In this way, Kong will serve new requests via the new
...
+
+```
+git diff 2.0.0 2.1.0 kong/templates/nginx_kong*.lua > kong_config_changes.diff
+```
+
+#### 3. Suggested Upgrade Path
+
+##### Upgrade from `0.x` to `2.1.0`
+
+The lowest version that Kong 2.1.0 supports migrating from is 1.0.0.
```
Comment (gszr) [inline UPGRADE.md:104]: Isn't it required to upgrade from Kong 1.0 to 1.5 before upgrading to 2.x?

File: UPGRADE.md (hunk 1 above)
Comment (hishamhm) [reply UPGRADE.md:70]: This was copied from the previous upgrade tests. Calling it "our openresty-build-tools repository" would be misleading. I'll try a different adjustement!

File: UPGRADE.md (hunk 4 above)
Comment (hishamhm) [reply UPGRADE.md:104]: Not really, as explained in the 2.0 section of this document. The 0.x migrations were squashed, but all 1.x-series migrations are there. We will squash the 1.x migrations on Kong 3.0.

File: UPGRADE.md (hunk 3 above)
Comment (hishamhm) [reply UPGRADE.md:43]: This was also copied from the previous versions. Are Kong docs standardizing on the Oxford comma? :)

Comment (gszr) [reply UPGRADE.md:43]: At least on the docs website I think we are :)

File: UPGRADE.md (hunk 2 above)
Comment (hishamhm) [reply UPGRADE.md]: Fixed!

File: UPGRADE.md (hunk 3 above)
Comment (hishamhm) [reply UPGRADE.md:43]: You're right! Fixing it!

https://github.com/Kong/docs.konghq.com/blob/master/STYLEGUIDE.md#serial-comma

Comment (hishamhm) [reply UPGRADE.md]: done!

**Pull request merged**